hypervisor/src/hypervisor.rs

```diff
@@ -118,3 +118,13 @@ pub trait Hypervisor: Send + Sync {
     #[cfg(feature = "tdx")]
     fn tdx_capabilities(&self) -> Result<TdxCapabilities>;
 }
+///
+/// Generic MemoryRegion struct
+///
+pub struct MemoryRegion {
+    pub slot: u32,
+    pub guest_phys_addr: u64,
+    pub memory_size: u64,
+    pub userspace_addr: u64,
+    pub flags: u32,
```

Comment:
You also need to define your own flags, otherwise you're leaking hypervisor implementation details.

Reply:
So I had a discussion with @jinankjain and @liuw, I am gonna be implementing read write and execute flags for the MemoryRegion struct. This is easily supported on mshv, but kvm only has a READONLY flag, so some flag configurations won't be valid(example, write is set, but read is not, this wont be possible to convert to kvm). I will add asserts to prevent these invalid flag configurations.